applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/usteer.js

```diff
@@ -144,22 +169,55 @@ function collectWlanAPInfos(compactconnectioninfo_table_entries, wlanAPInfos) {
 	}
 };
 
+var callNetworkRrdnsLookup = rpc.declare({
+	object: 'network.rrdns',
+	method: 'lookup',
+	params: [ 'addrs', 'timeout', 'limit' ],
+	expect: { '': {} }
+});
+
+
 function collectRemoteHosts (remotehosttableentries,Remotehosts) {
+	var ipAddrs = [];
 	for (var IPaddr in Remotehosts) {
-			remotehosttableentries.push([IPaddr, Remotehosts[IPaddr]['id']]);
-		}
+		if (typeof dns_cache[IPaddr] === 'undefined') ipAddrs.push(IPaddr);
+	}
+
+	L.resolveDefault(callNetworkRrdnsLookup(ipAddrs, 1000, 1000), {}).then(function(replies) {
+				for (var index in ipAddrs) {
```

Comment:
Is `index` an enumerable property of `ipAddrs`? Or an iterable object? The latter, I think, should use:

```js
				for (var index of ipAddrs) {
```

e.g.
```js
let list = [4, 5, 6];

for (let i in list) {
   console.log(i); // "0", "1", "2",
}

for (let i of list) {
   console.log(i); // "4", "5", "6"
}
```

In this case, simply:
```js
for (var ip of ipAddrs) {
  if (!ip)
    continue;
  if (replies[ip]) {
...
```
```

Reply:
Ah yes, using "of" is indeed better